### PR TITLE
Download updates from registers

### PR DESCRIPTION
**Pull request by karlbaker02**
### Context

This PR introduces the ability for the client library to download only the updates from a register it has previously downloaded data for, rather than needing to download the entire register and re-parse it all again. To achieve this, we are using the latest `user` entry number currently read up-to by the last download, then downloading RSF from the register from this point.

### Changes proposed in this pull request

Update register data by downloading partial RSF, not full RSF.

### Guidance to review

The register data stored by the `RegisterClient` has been moved to an instance variable, due to a limitation of the MiniCache gem we are using which will wipe the data it has for a key if the cache expiration is deemed to have expired. Instead, we continue to use the MiniCache gem, but we use it as a means of controlling access to the underlying `@data` instance variable, so that it can continue to be updated when the cache expires, but without the side effects of wiping all data currently held and storing register data twice.

### CODE REVIEW

File: .version [hunk 1]
```diff
@@ -1 +1 @@
-0.4.0
\ No newline at end of file
+0.5.0
```
Comment (gidsg) [inline .version]: Do we need a major version jump here, as I don't *think* we are changing the public API?

File: lib/register_client.rb [hunk 2]
```diff
@@ -99,69 +107,63 @@ def get_expired_records
...
+      @data
     end
 
-    def parse_rsf(rsf)
-      items = {}
-      entries = { user: [], system: [] }
-      records = { user: {}, system: {} }
-      user_entry_number = 1
-      system_entry_number = 1
+    def download_rsf(register, phase, start_entry_number)
```
Comment (gidsg) [inline lib/register_client.rb]: Would it make sense to default `start_entry_number` to 0? Then the tests for this method wouldn't need to parse that argument.

File: .version (hunk 1 above)
Comment (karlbaker02) [reply .version]: Well, technically this is a minor version jump according to semver.

File: lib/register_client.rb (hunk 2 above)
Comment (karlbaker02) [reply lib/register_client.rb]: Ah yes, sorry, that should definitely be zero, will update.

Comment (karlbaker02) [reply lib/register_client.rb]: I think I misinterpreted your comment - we could make it a default value of zero but for the sake of only the tests, I don't see the need for this.

**Pull request merged**